### PR TITLE
Use SliceableDataset

### PR DESCRIPTION
**Pull request by Hakuyume**
merge after #571

### CODE REVIEW

**Comment by Hakuyume** [comment]
How about this style?

image

**Comment by yuyu2172** [comment]
I prefer setting the range of `label` to `[0, n_class]`. You already use variables such as `H` and `W`, so I think it is fine to use a variable.

**Comment by Hakuyume** [comment]
Should it be `n_fg_class`?

**Comment by yuyu2172** [comment]
Yes

File: chainercv/datasets/ade20k/ade20k_test_image_dataset.py [hunk 1]
```diff
@@ -26,27 +24,27 @@ class ADE20KTestImageDataset(dataset.DatasetMixin):
             dataset is automatically downloaded into
             :obj:`$CHAINER_DATASET_ROOT/pfnet/chainercv/ade20k`.
 
+    This dataset returns the following data.
```
Comment (yuyu2172) [inline chainercv/datasets/ade20k/ade20k_test_image_dataset.py]: This line is not necessary.

File: chainercv/datasets/sbd/sbd_instance_segmentation_dataset.py [hunk 2]
```diff
@@ -39,9 +36,22 @@ class SBDInstanceSegmentationDataset(chainer.dataset.DatasetMixin):
...
+    This dataset returns the following data.
+
+    .. csv-table::
+        :header: name, shape, dtype, format
+
+        :obj:`img`, ":math:`(3, H, W)`", :obj:`float32`, \
+        "RGB, :math:`[0, 255]`"
+        :obj:`mask`, ":math:`(R, H, W)`", :obj:`bool`, --
+        :obj:`label`, ":math:`(R,)`", :obj:`int32`, \
+        ":math:`[0, 19]`"
```
Comment (yuyu2172) [inline chainercv/datasets/sbd/sbd_instance_segmentation_dataset.py]: n_class

**Pull request merged**